File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-evcxr/src/test/java/de/invesdwin/scripting/rust/runtime/evcxr/SimpleCallbackTest.rs
println!("putUuid: {}", putUuid);

let getSecretStaticCallback: String = callback("getSecretStatic", &[param(putUuid)]);
println!("getSecretStaticCallback: {}", getSecretStaticCallback);

let getSecretCallback: String = callback("getSecret", &[param(putUuid)]);
println!("getSecretCallback: {}", getSecretCallback);

let getSecretExpressionCallback: String = callback("getSecretExpression", &[param(putUuid)]);
println!("getSecretExpressionCallback: {}", getSecretExpressionCallback);

// callback_void or callback_dynamic would be slightly more efficient for void methods, but callback with unit type also works fine
callback::<()>("voidMethod", &[]);

let callManyParams: f64 = callback("callManyParams", &[
    param(true),
    param(2),
    param(3),
    param("4"),
    param(5),
    param(6),
    param(7.0),
    param(8.0),
    param("123456789"),
    param(10.0)
]);
println!("callManyParams: {}", callManyParams);
if (callManyParams != 55.0) {
    panic!("callManyParams unexpected result: {}", callManyParams);
}

let callManyParamsExpression: f64 = callback("callManyParamsExpression", &[
    param(true),
    param(2),
    param(3),
    param("4"),
    param(5),
    param(6),
    param(7.0),
    param(8.0),
    param("123456789"),
    param(10.0)
]);
println!("callManyParamsExpression: {}", callManyParamsExpression);
if (callManyParamsExpression != 55.0) {
    panic!("callManyParamsExpression unexpected result: {}", callManyParamsExpression);
}

let callManyParamsExpressionMultiline: f64 = callback("callManyParamsExpressionMultiline", &[
    param(true),
    param(2),
    param(3),
    param("4"),
    param(5),
    param(6),
    param(7.0),
    param(8.0),
    param("123456789"),
    param(10.0)
]);
println!("callManyParamsExpressionMultiline: {}", callManyParamsExpressionMultiline);
if (callManyParamsExpressionMultiline != 55.0) {
    panic!("callManyParamsExpressionMultiline unexpected result: {}", callManyParamsExpressionMultiline);
}

let getManyParamsExpression = putManyParamsExpression;
println!("getManyParamsExpression: {}", getManyParamsExpression);
let getManyParamsExpressionMultiline = putManyParamsExpressionMultiline;
println!("getManyParamsExpressionMultiline: {}", getManyParamsExpressionMultiline);

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/SimpleCallbackTest.lua
print("putUuid")
print(putUuid)

SimpleCallbackTest = java.import("de.invesdwin.scripting.runtime.lua.tests.callback.SimpleCallbackTest")
getSecretStaticImport = SimpleCallbackTest:getSecretStatic(putUuid)
print("getSecretStaticImport")
print(getSecretStaticImport)

getSecretStaticCallback = callback("getSecretStatic", putUuid)
print("getSecretStaticCallback")
print(getSecretStaticCallback)

getSecretCallback = callback("getSecret", putUuid)
print("getSecretCallback")
print(getSecretCallback)

getSecretExpressionCallback = callback("getSecretExpression", putUuid)
print("getSecretExpressionCallback")
print(getSecretExpressionCallback)

callback("voidMethod")

callManyParams = callback("callManyParams", true, 2, 3, '4', 5, 6, 7.0, 8.0, "123456789", 10.0)
if(callManyParams ~= 55) then
	error("callManyParams unexpected result: " .. callManyParams)
end
callManyParamsExpression = callback("callManyParamsExpression", true, 2, 3, '4', 5, 6, 7.0, 8.0, "123456789", 10.0)
if(callManyParamsExpression ~= 55) then
	error("callManyParamsExpression unexpected result: " .. callManyParamsExpression)
end
callManyParamsExpressionMultiline = callback("callManyParamsExpressionMultiline", true, 2, 3, '4', 5, 6, 7.0, 8.0, "123456789", 10.0)
if(callManyParamsExpressionMultiline ~= 55) then
	error("callManyParamsExpressionMultiline unexpected result: " .. callManyParamsExpressionMultiline)
end

getManyParamsExpression = putManyParamsExpression
print("getManyParamsExpression")
print(getManyParamsExpression)
getManyParamsExpressionMultiline = putManyParamsExpressionMultiline
print("getManyParamsExpressionMultiline")
print(getManyParamsExpressionMultiline)
